src/server/frontend_wayland/wl_client.h

```diff
@@ -52,7 +56,7 @@ class WlClient : public wayland::LifetimeTracker
         std::shared_ptr<SessionAuthorizer> const& session_authorizer,
         std::function<void(WlClient&)>&& client_created_callback);
 
-    static auto from(wl_client* client) -> WlClient*;
+    static auto from(wl_client* client) -> WlClient&;
```

Comment:
I wouldn't have changed this signature, just document the post-condition.

Reply:
Makes the expectation a little more explicit